### PR TITLE
Add more clarify on how reclaimWithinCohort interacts with fair sharing

### LINKED ISSUE #3286: Document how preemption fields interact with fair sharing strategies
**What would you like to be added**:

Clarify the semantics of `reclaimWithinCohort` and `withinClusterQueue` when using fair sharing and how it affects choosing candidates for preemption.

/kind documentation

**Why is this needed**:

The documentation leaves many open questions

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [x] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by avrittrohwer**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

1. Clarifies API docs for reclaimWithinCohort 'if the pending Workload fits within the nominal quota' wording.
2. Makes it more obvious in concept docs that fair sharing works in addition to the preemption policies set.

#### Which issue(s) this PR fixes:

Fixes #3286

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: site/content/en/docs/reference/kueue.v1beta1.md
Comment (mbobrovskyi) [inline site/content/en/docs/reference/kueue.v1beta1.md:1]: This is  generated page and changes will be removed after `make verify`.

Comment (alculquicondor) [reply site/content/en/docs/reference/kueue.v1beta1.md:1]: Right, you have to make the changes in https://github.com/kubernetes-sigs/kueue/blob/main/apis/kueue/v1beta1/clusterqueue_types.go and run `make manifests generate update-helm generate-apiref`

File: site/content/en/docs/reference/kueue.v1beta1.md [hunk 1]
```diff
@@ -720,10 +720,12 @@ Workloads from other ClusterQueues in the cohort that are using more than
 their nominal quota. The possible values are:</p>
 <ul>
 <li><code>Never</code> (default): do not preempt Workloads in the cohort.</li>
-<li><code>LowerPriority</code>: if the pending Workload fits within the nominal
+<li><code>LowerPriority</code>: **Classic preemption** if the pending Workload fits within the nominal
 quota of its ClusterQueue, only preempt Workloads in the cohort that have
-lower priority than the pending Workload.</li>
-<li><code>Any</code>: if the pending Workload fits within the nominal quota of its
+lower priority than the pending Workload. **Fair sharing** only preempt
+Workloads in the cohort that have lower priority then the pending Workload and
```
Comment (alculquicondor) [inline site/content/en/docs/reference/kueue.v1beta1.md:726]:
```suggestion
Workloads in the cohort that have lower priority than the pending Workload and
```

File: site/content/en/docs/concepts/preemption.md [hunk 2]
```diff
@@ -117,6 +117,11 @@ admitted when accounting back the quota usage of the target Workload.
 
 ## Fair Sharing
 
+Fair sharing introduces the concepts of ClusterQueue share values and preemption
+strategies. These work together with the preemption policies set in
+`withinClusterQueue` and `reclaimWithinCohort` to determine if pending workloads
+can preempt admitted workloads.
```
Comment (alculquicondor) [inline site/content/en/docs/concepts/preemption.md:123]: Maybe let's add: The goal of fair sharing is to achieve an equal or weighted share of the borrowable resources among the tenants of a cohort.

File: site/content/en/docs/reference/kueue.v1beta1.md
Comment (avrittrohwer) [reply site/content/en/docs/reference/kueue.v1beta1.md:1]: ack, updated

File: site/content/en/docs/concepts/preemption.md (hunk 2 above)
Comment (avrittrohwer) [reply site/content/en/docs/concepts/preemption.md]: added

**Comment by mimowo** [comment]
Thanks!
/lgtm
/approve

**Comment by mimowo** [comment]
/cherry-pick website
As it might be useful for some users before 0.9 is released.

**Comment by mimowo** [comment]
@avrittrohwer could you check if the conflicts are easily fixable, if so, then feel free to create the cherry-pick manually. You can use the script: `./hack/cherry_pick_pull.sh` for that.

**Comment by avrittrohwer** [comment]
> @avrittrohwer could you check if the conflicts are easily fixable, if so, then feel free to create the cherry-pick manually. You can use the script: `./hack/cherry_pick_pull.sh` for that.

https://github.com/kubernetes-sigs/kueue/pull/3294

**Pull request merged**